### PR TITLE
Integration test

### PR DESCRIPTION
**Pull request by yonas44**
## These are the changes implemented in this PR:
- Make sure that the N+1 problem is solved when fetching all posts.
- Use `Capybara` to write integration tests.
- Write integration test of `User#index` template.
- Write integration test of `User#show` template.
- Write integration test of `Post#index` template.
- Write integration test of `Post#show` template.

### Before: 
Screenshot from 2023-03-07 12-54-55

### After: 
Screenshot from 2023-03-07 12-55-18

### CODE REVIEW

File: spec/feature/post_system_spec.rb [hunk 1]
```diff
@@ -0,0 +1,71 @@
+require 'rails_helper'
+
```
Comment (rloterh) [inline spec/feature/post_system_spec.rb:2]: - [ ] Your integration specs do not assert most of the test criteria specified for both _user_ and _post_ views. Let's ensure that our test case asserts **all** given criteria on the stated on the [project requirement page](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md).

Comment (rloterh) [inline spec/feature/post_system_spec.rb:2]: - [ ] Some of the integration specs are failing! Please ensure **all** specs are passing.

**Failing specs:**
image

Comment (yonas44) [reply spec/feature/post_system_spec.rb:2]: Hello @rloterh thank you for the feedback, the problem was for the test to work the test-database is supposed to be populated with the data from the `seed.rb` using the command `rails db:seed RAILS_ENV=test` in the terminal and I forgot to mention that in my discription.

Comment (rloterh) [reply spec/feature/post_system_spec.rb:2]: Hi @yonas44 

I actually implemented all of the steps you have mentioned!

**Review by ShadyShawkat — APPROVED** [review]
## Hi @yonas44, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- All linters are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**